common/cpp_unit_test_runner/src/build_emscripten.mk

```diff
@@ -0,0 +1,74 @@
+# Copyright 2020 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This file contains the implementation of the ../include.mk interface that
+# builds the C++ unit test runner using the Emscripten toolchain.
+
+# Flags passed to the Emscripten compiler/linker tools in test builds.
+#
+# Explanation:
+# MODULARIZE: Disable putting the Emscripten module JavaScript loading code into
+#   a factory function, so that the test runner module is loaded automatically
+#   on startup.
+TEST_ADDITIONAL_EMSCRIPTEN_FLAGS := \
+	-s MODULARIZE=0 \
```

Comment:
Do we expect that all emscripten specific flags are applied to both compiling and linking? If not, it may be warranted to split this variable into two, or just use it directly in the two instances where it is used (though this solution does provide a nice place for the comment about MODULARIZE).

Reply:
I'm not very experienced in Emscripten, but I was following their guideline that recommends using the same arguments: https://github.com/emscripten-core/emscripten/blob/9d37058cedb5ffa482202eb50812e67cd9bec810/src/settings.js#L30
I already experienced it with things like `-pthread` which result in weird compilation issues when it's missing somewhere. I can imagine some other flags can even cause runtime issues if used inconsistently.

Having said that, we definitely can move a flag into a specific location (like CXXFLAGS or LDFLAGS), as long as we 100% sure that it's needed only on a specific stage. But I haven't found any reference source for that yet.